src/luaState.cpp

```diff
@@ -157,19 +197,106 @@ Variant LuaState::getVar(int index) const {
 
 // Pull a global variant from Lua to GDScript
 Variant LuaState::pullVariant(String name) {
-	lua_getglobal(L, name.ascii().get_data());
+#ifndef LAPI_LUAJIT
+	lua_pushglobaltable(L);
+#else
+	lua_pushvalue(L, LUA_GLOBALSINDEX);
+#endif
+#ifndef LAPI_GDEXTENSION
+	Vector<String> strs = name.split(".");
+#else
+	PackedStringArray strs = name.split(".");
+#endif
+	for (String str : strs) {
+		if (lua_type(L, -1) != LUA_TTABLE) {
+			lua_pop(L, 1); // Please do not do unforeseen consequences thanks.
+			lua_pushnil(L);
+			break;
+		}
+		lua_getfield(L, -1, str.ascii().get_data());
+		lua_remove(L, -2);
+	}
```

Comment:
One other thing, this same code snippet is used a lot, it would probably be better served as a utility function that can take a state and Variable name, and put the variable on the top of the stack.